allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/behavior_neuropixels_project_cloud_api.py

```diff
@@ -0,0 +1,149 @@
+import pandas as pd
+
+from allensdk.brain_observatory.behavior.behavior_project_cache.\
+    project_apis.data_io.project_cloud_api_base import ProjectCloudApiBase  # noqa: E501
+
+from allensdk.brain_observatory.behavior.behavior_session import (
+    BehaviorSession)
+
+from allensdk.brain_observatory.ecephys.behavior_ecephys_session \
+    import BehaviorEcephysSession
+
+
+class VisualBehaviorNeuropixelsProjectCloudApi(ProjectCloudApiBase):
+
+    MANIFEST_COMPATIBILITY = ["0.1.0", "1.0.0"]
+
+    def _load_manifest_tables(self):
+
+        self._get_ecephys_session_table()
+        self._get_behavior_session_table()
+        self._get_unit_table()
+        self._get_probe_table()
+        self._get_channel_table()
+
+    def get_behavior_session(
+            self, behavior_session_id: int) -> BehaviorSession:
+        """
+        Since we are not releasing behavior-only NWB files
+        with the VBN June 2022 release, the behavior sesion data
+        is obtained from the behavior_ecephys_session NWB file
+        based on the associated behavior_ecephys_session_id
+
+        Parameters
+        ----------
+        behavior_session_id: int
+            the id of the behavior_session
+
+        Returns
+        -------
+        BehaviorSession
+
+        Notes
+        -----
+        behavior session does not include file_id.
+        The file id is accessed via ecephys_session_id key
+        from the ecephys_session_table
+        """
+        row = self._behavior_session_table.query(
+                f"behavior_session_id=={behavior_session_id}")
+        if row.shape[0] != 1:
+            raise RuntimeError("The behavior_session_table should have "
+                               "1 and only 1 entry for a given "
+                               "behavior_session_id. For "
+                               f"{behavior_session_id} "
+                               f" there are {row.shape[0]} entries.")
+        row = row.squeeze()
+        ecephys_session_id = int(row.ecephys_session_id)
+
+        row = self._ecephys_session_table.query(f"index=={ecephys_session_id}")
+
+        if len(row) == 0:
+            raise RuntimeError(f"ecephys_session: {ecephys_session_id} "
+                               f"corresponding to "
+                               f"behavior_session: {behavior_session_id} "
+                               f"does not exist in the ecephys_session_table ")
+
+        file_id = str(int(row[self.cache.file_id_column]))
+        data_path = self._get_data_path(file_id=file_id)
+
+        return BehaviorSession.from_nwb_path(str(data_path))
```

Comment:
I think this should return the `VBNBehaviorSession` class defined in `brain_observatory/ecephys/behavior_ecephys_session.py`.

I know that the `from_nwb` logic is the same for both `BehaviorSession` and `VBNBehaviorSession`, but, **if** we ever need to change the `from_nwb` logic for one and not the other, VBN-related changes will be in `VBNBehaviorSession`.

This is not a strong ask. This is only my opinion.

Reply:
still not sure about this one. I have kept it as is.